stack/rush-stack-compiler-2.9/src/TslintRunner.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { ITerminalProvider } from '@microsoft/node-core-library';
+import * as TSLint from 'tslint';
+
+import { CmdRunner } from './CmdRunner';
+import { ToolPaths } from './ToolPaths';
+import { RushStackCompilerBase } from './RushStackCompilerBase';
+
+export type WriteFileIssueFunction = (
+  filePath: string,
+  line: number,
+  column: number,
+  errorCode: string,
+  message: string
+) => void;
+
+/**
+ * @public
+ */
+export interface ITslintRunnerConfig {
+  fileError: WriteFileIssueFunction;
+  fileWarning: WriteFileIssueFunction;
+
+  /**
+   * If true, displays warnings as errors. Defaults to false.
+   */
+  displayAsError?: boolean;
+}
+
+/**
+ * @beta
+ */
+export class TslintRunner extends RushStackCompilerBase<ITslintRunnerConfig> {
+  private _cmdRunner: CmdRunner;
+
+  constructor(taskOptions: ITslintRunnerConfig, rootPath: string, terminalProvider: ITerminalProvider) {
+    super(taskOptions, rootPath, terminalProvider);
+    this._cmdRunner = new CmdRunner(
+      this._standardBuildFolders,
+      this._terminal,
+      {
+        packagePath: ToolPaths.tslintPackagePath,
+        packageJson: ToolPaths.tslintPackageJson,
+        packageBinPath: path.join('bin', 'tslint')
+      }
+    );
+  }
+
+  public invoke(): Promise<void> {
+    const args: string[] = [
+      '--format', 'json',
+      '--project', this._standardBuildFolders.projectFolderPath
+    ];
+
```

Comment:
Hrmmm.... this is a LOT of copy+pasting.  How are we supposed to keep these files in sync?